server/data/prisonerSearchClient.test.ts

```diff
@@ -57,4 +52,57 @@ describe('prisonerSearchClient', () => {
       expect(nock.isDone()).toBe(true)
     })
   })
+
+  describe('getPrisonersByPrisonId', () => {
+    it('should get prisoners by prison id given prison ID exists', async () => {
+      // Given
+      const prisonId = 'BXI'
+      const systemToken = 'a-system-token'
+
+      const page = 0
+      const pageSize = 100
+
+      const pagedCollectionOfPrisoners = aValidPagedCollectionOfPrisoners({
+        number: 2,
+        content: [
+          aValidPrisoner({ firstName: 'Fred', lastName: 'Blogs' }),
+          aValidPrisoner({ firstName: 'Martin', lastName: 'McDougal' }),
+        ],
+      })
+      prisonerSearchApi
+        .get(`/prison-search/prison/${prisonId}?page=${page}&size=${pageSize}`)
+        .reply(200, pagedCollectionOfPrisoners)
+
+      // When
+      const actual = await prisonerSearchClient.getPrisonersByPrisonId(prisonId, page, pageSize, systemToken)
+
+      // Then
+      expect(actual).toEqual(pagedCollectionOfPrisoners)
+      expect(nock.isDone()).toBe(true)
+    })
+
+    it('should get zero prisoners by prison id given prison ID does not exist', async () => {
```

Comment:
This test demonstrates that we expect a response with an empty collection, rather than a 404
Functionally its not massively different from the test above, but it does demonstrate the intent and behaviour, especially as its not our API 👍